mitodl/mit-open
# Removing nginx serving of the frontend for local setups

### What are the relevant tickets?




Closes https://github.com/mitodl/hq/issues/4602
Closes https://github.com/mitodl/hq/issues/4611


### Description (What does it do?)

- All nginx serving of frontend is removed (except for the e2e tests as a special case).
- Updated the nginx setup in docker-compose to use the same nginx.conf that production uses. This implementation was lifted from [MITx Online](https://github.com/mitodl/mitxonline/tree/main/nginx)
- Changed the configuration of `env_file` so that it's split up between frontend vs. backend so it's generally cleaner and not as noisy.
- Renamed the environment variables for the API and app urls so they have more obvious names.

### How can this be tested?


- You should be able to still use your existing `.env` file as I left support for that in place
- You can also split it up into `env/frontend.local.env` and `env/backend.local.env`. There's also `env/shared.local.env`, but that is mainly for the `MITOPEN_API_BASE_URL` and `MITOPEN_APP_BASE_URL` vars.
- You can also remove a lot of vars from your `.env` file(s) that are now defined by default in the checked in `.env` files. A lot of time we just ended up leaving what was in `.env` and this should hopefully make that file a lot less noisy.
- You should be able to do all the normal things you can do:
  - Run the app via `docker compose up` with the different combinations of `COMPOSE_PROFILES` set.
    - If you're running both profiles, you'll now access the app in your browser via port `8062` (e.g. http://open.odl.local:8062/). The API and django-admin are still on port `8063`.
  - Run one-off containers via `docker compose run --rm web bash`, etc.
  - Run the tests on both frontend and backend.
